crates/biome_js_analyze/src/lint/style/no_restricted_imports.rs

```diff
@@ -315,6 +341,201 @@ declare_lint_rule! {
     /// ```js,use_options
     /// import { Bar } from 'import-bar';
     /// ```
+    ///
+    /// ### `patterns`
+    ///
+    /// This option allows you to specify multiple modules to restrict using gitignore-style patterns or regular expressions.
+    ///
+    /// ```json,options
+    /// {
+    ///    "options": {
+    ///         "patterns": ["utils/*", "!utils/foo"]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// #### Invalid
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// import bar from "utils/bar";
+    /// ```
+    ///
+    /// #### Valid
+    ///
+    /// ```js,use_options
+    /// import foo from "utils/foo";
+    /// ```
+    ///
+    /// ### `group`
+    ///
+    /// The patterns array can also include objects. The group property is used to specify the gitignore-style patterns for restricting modules and the message property is used to specify a custom message.
+    /// **`group` cannot be used in combination with `regex`.**
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "patterns": [{
+    ///             "group": ["import-foo/*", "!import-foo/bar"],
+    ///             "message": "import-foo is deprecated, except the modules in import-foo/bar."
+    ///         }]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// #### Invalid
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// import foo from 'import-foo/foo';
+    /// ```
+    ///
+    /// #### Valid
+    ///
+    /// ```js,use_options
+    /// import foo from 'import-foo';
+    /// import bar from 'import-foo/bar';
+    /// ```
+    ///
+    /// ### `regex`
+    ///
+    /// The regex property is used to specify the regex patterns for restricting modules.
+    /// **`regex` cannot be used in combination with `group`.**
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///        "patterns": [{
+    ///             "regex": "^import-foo/\\d{4}-\\d{2}-\\d{2}$",
+    ///             "message": "import-foo/ + particular date format is deprecated."
+    ///         }]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// #### Invalid
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// import { foo } from 'import-foo/2025-05-17';
+    /// ```
+    ///
+    /// #### Valid
+    ///
+    /// ```js,use_options
+    /// import { foo } from 'import-foo/foo';
+    /// ```
+    ///
+    /// ### `importNames`
+    ///
+    /// You can also specify importNames within objects inside the patterns array. In this case, the specified names apply only to the associated group or regex property.
+    /// **`importNames` cannot be used in combination with `importNamePattern` , `allowImportNames` and `allowImportNamePattern`.**
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///        "patterns": [{
+    ///             "group": ["utils/*"],
```

Comment:
I wonder if we should rename `group` by `globs`. This look more intuitive to me. Any opinion?